src/buildstream/_elementsourcescache.py

```diff
@@ -195,7 +197,7 @@ def _get_source(self, ref):
                 source_proto.ParseFromString(f.read())
                 return source_proto
         except FileNotFoundError as e:
-            raise SourceCacheError("Attempted to access unavailable source: {}".format(e)) from e
+            raise SourceCacheError("Attempted to access unavailable source: {}".format(e), temporary=True) from e
```

Comment:
If I understand correctly, here we are about to push sources (this is called from `self._push_source()`) and we call `_get_source()` to get the proto from local CAS.

I suppose this error could occur if for some reason the local source has been purged/expired from local CAS before pushing.

I don't think this error can possibly recover with retries, so `temporary` is probably wrong in this case.

Reply:
Thanks, will remove